src/helpers/generate-matrix.ts

```diff
@@ -17,11 +19,42 @@ import { chunk } from 'lodash';
 export class GenerateMatrix extends HelperInputs {
   paths = '';
   batches?: string;
+  load_balancing_sizes?: string;
 }
 
-export const generateMatrix = ({ paths, batches = '1' }: GenerateMatrix) => {
+export const generateMatrix = ({ paths, batches: _batches = '1', load_balancing_sizes }: GenerateMatrix) => {
   const matrixValues = paths.split(/[\n,]/);
-  return {
-    include: chunk(matrixValues, Math.ceil(matrixValues.length / Number(batches))).map(chunk => ({ path: chunk.join(',') }))
-  };
+  const batches = Number(_batches);
+  if (!load_balancing_sizes) {
+    return {
+      include: chunk(matrixValues, Math.ceil(matrixValues.length / batches)).map(chunk => ({ path: chunk.join(',') }))
+    };
+  } else {
+    const loadBalancingSizes = load_balancing_sizes.split(/[\n,]/).map(size => Number(size));
+    if (loadBalancingSizes.length !== matrixValues.length) throw new Error('load_balancing_sizes have the same length as paths');
+    const targetLoadSize = loadBalancingSizes.reduce((acc, size) => acc + size, 0) / batches;
+    const loadBalancedPaths: string[] = [];
+    let currentLoadSize = 0;
+    let currentBatch: string[] = [];
+    matrixValues.forEach((path, index) => {
```

Comment:
Wonder if we can map the `matrixValues` to `loadBalancedPaths` rather than iterating and mutating

Reply:
a more functional approach would be to use `reduce` instead of `forEach`, but my worry was that this would be less readable/maintainable. i don't see a great way to compute the result in-place but it could be done with an in-place `number[][]` of loads and a separate `string[]` of resulting paths initialized before the `reduce` block. then there wouldn't be any reassignment